tools/pymanifold.cpp

```diff
@@ -0,0 +1,169 @@
+#include "manifold.h"
+#include "meshIO.h"
+#include "pybind11/functional.h"
+#include "pybind11/numpy.h"
+#include "pybind11/operators.h"
+#include "pybind11/pybind11.h"
+#include "pybind11/stl.h"
+
+namespace py = pybind11;
+
+using namespace manifold;
+
+void exportManifold(Manifold &m, std::string name) {
+  Mesh out = m.GetMesh();
+  manifold::ExportOptions options;
+  options.faceted = true;
+  options.mat.roughness = 0.2;
+  options.mat.metalness = 0.0;
+  manifold::ExportMesh(name, out, options);
+};
+
+typedef std::tuple<float, float> Float2;
+typedef std::tuple<float, float, float> Float3;
+
+struct PolygonsWrapper {
+  std::unique_ptr<Polygons> polygons;
+};
+
+PYBIND11_MODULE(pymanifold, m) {
+  m.doc() = "Python binding for the manifold library. Please check the C++ documentation for APIs.\n"
+      "This binding will perform copying to make the API more familiar to OpenSCAD users.";
+  py::class_<Manifold>(m, "Manifold")
+      .def(py::init<>())
+      .def(py::init([](std::vector<Manifold> &manifolds) {
+        Manifold result;
+        for (Manifold &manifold : manifolds)
+          result += manifold;
+        return result;
+      }), "Construct manifold as the union of a set of manifolds.")
+      .def(py::self + py::self)
+      .def(py::self - py::self)
+      .def(py::self ^ py::self)
+      .def("transform",
+           [](Manifold &self, py::array_t<float> &mat) {
+             auto mat_view = mat.unchecked<2>();
+             if (mat_view.shape(0) != 3 || mat_view.shape(1) != 4)
+               throw std::runtime_error("Invalid matrix shape");
+             glm::mat4x3 mat_glm;
+             for (int i = 0; i < 3; i++) {
+               for (int j = 0; j < 4; j++) {
+                 mat_glm[j][i] = mat_view(i, j);
+               }
+             }
+             return self.Transform(mat_glm);
+           })
+      .def(
+          "translate",
+          [](Manifold self, float x = 0.0f, float y = 0.0f, float z = 0.0f) {
```

Comment:
Whereas `translate`, `scale`, `rotate` are doing a copy now? I built Manifold to do these operations lazily already, so you could string many together and it would apply them all at once only when needed. I feel like doing copies between each operation, especially when there could be a long string of them before the next Boolean, would be a pretty major drag on performance.

Reply:
Indeed, but as I've said we can use something like a `shared_ptr` to share the underlying manifold and do CoW. Will implement that if needed when the above discussion about the API settles.

Reply:
Ah, I'm starting to understand this better. That's an interesting idea. It still seems like it would be preferable to keep the API consistent among the different language bindings. Would you propose we change the C++ API to function this way as well? This project is young, so better to make breaking API changes now than when there are more users.

Reply:
Yes, it would be ideal to chang the C++ API as well. In fact the prototype implementation `ManifoldSet` in my fork contains some code duplication for handling transforms, so it would be nice to integrate it to the Manifold class directly.

Reply:
Okay, we have two sort of parallel discussions going here about the API and copying. I'd like to unblock this PR. How about we stick with the API you have for now, and look at modifying the C++ API to match in a following PR? Do you want to go ahead and get the tests passing and get this merged?